projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/assets/class-verbum-block-utils.php

```diff
@@ -20,19 +20,60 @@ public static function remove_blocks( $content ) {
 			return $content;
 		}
 
-		$allowed_blocks = self::get_allowed_blocks();
 		// The block attributes come slashed and `parse_blocks` won't be able to parse them.
 		$content = wp_unslash( $content );
-		$blocks  = parse_blocks( $content );
-		$output  = '';
+
+		// Parse blocks first
+		$blocks = parse_blocks( $content );
+
+		// Filter blocks recursively
+		$filtered_blocks = self::filter_blocks_recursive( $blocks );
+
+		// Convert the filtered blocks back to HTML
+		return serialize_blocks( $filtered_blocks );
+	}
+
+	/**
+	 * Recursively filter blocks and their inner blocks
+	 *
+	 * @param array $blocks Array of blocks to filter.
+	 * @return array Filtered blocks.
+	 */
+	private static function filter_blocks_recursive( $blocks ) {
```

Comment:
It feels like this shouldn't be so complicated, am I missing a convenience function or a better way of doing things?

Reply:
Thanks for checking. I guess it is what it is, I'll go ahead with this.